test/UnitTests/MSTest.Analyzers.UnitTests/AvoidAssertAreSameWithValueTypesAnalyzerTests.cs

```diff
@@ -189,4 +189,51 @@ public void TestMethod()
 
         await VerifyCS.VerifyCodeFixAsync(code, fixedCode);
     }
+
+    [TestMethod]
+    public async Task AnalyzerMessageShouldBeCorrect()
+    {
+        string code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            using System.Collections.Generic;
+
+            [TestClass]
+            public class MyTestClass
+            {
+                [TestMethod]
+                public void TestMethod()
+                {
+                    {|#0:Assert.AreSame(0, 0)|};
+                    {|#1:Assert.AreNotSame(0, 1)|};
+                }
+            }
+            """;
+
+        string fixedCode = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            using System.Collections.Generic;
+
+            [TestClass]
+            public class MyTestClass
+            {
+                [TestMethod]
+                public void TestMethod()
+                {
+                    Assert.AreEqual(0, 0);
+                    Assert.AreNotEqual(0, 1);
+                }
+            }
+            """;
+
+        await new VerifyCS.Test()
+        {
+            TestCode = code,
+            FixedCode = fixedCode,
+            ExpectedDiagnostics =
+            {
+                VerifyCS.Diagnostic().WithLocation(0).WithMessage("Use 'AreEqual' instead of 'AreSame' when comparing value types"),
+                VerifyCS.Diagnostic().WithLocation(1).WithMessage("Use 'AreNotEqual' instead of 'AreNotSame' when comparing value types"),
```

Comment:
Let's use `WithArguments` instead of `WithMessage` so that the test still works when using different culture.

Reply:
```suggestion
                VerifyCS.Diagnostic().WithLocation(0).WithArguments("AreEqual", "AreSame"),
                VerifyCS.Diagnostic().WithLocation(1).WithArguments("AreEqual", "AreSame"),
```

Reply:
It makes it less clear to see from the test that the message is correct, but I agree that it's better to make the test run under different culture.